app/helpers/spree/admin/slides_helper.rb

```diff
@@ -14,6 +14,16 @@ def get_image_link_by_type(slide, type)
 
         return '----'
       end
+
+      def product_options_for_select(slide)
+        if slide.product_id.present?
+          Spree::Product.where(id: slide.product_id).to_a
```

Comment:
TODO: validate with multi vendor and multi store..

@sushil-truemark can you please also help me understand what this helper is doing?

Reply:
This helper is use for listing down the options for the products. (specially for displaying product in edit slide)
Below is over all use case
1. When creating the slide, the value remain nil because product will be fetch only after user search the product name or slug.
2. After creating the slide, we needed to fetch product which is selected by user in edit section of slide. 

Since we are only fetching the product which is preselected by user. There couldn't be cause where it will affect multi vendor and multi store.

Reply:
Spree::Product
there could be issue because you are directly calling from model and this skips any scope that a user is limited to..

so the data that is not intended to be shared with the user doing search might be revealed

Reply:
This helper is use for listing down the options for the products. (specially for displaying product in edit slide)

I am still confused..I am expecting the data to come from controller..
It is not a good practice to fetch data from view

Reply:
Could you check it once if latest changes avoid such scenario?
we are fetching data based of association

Reply:
we are getting data from controller.

> I am still confused..I am expecting the data to come from controller..

Reason, why we added such scenario, we need to add at least selected product in options list. 
<img width="1121" alt="image" src="https://github.com/spreehood/spree_slider/assets/133846865/67198137-4500-4187-9bdb-83020573b4f9">